packages/framework/fluid-static/src/fluidContainer.ts

```diff
@@ -178,7 +182,7 @@ export interface IFluidContainer extends IEventProvider<IFluidContainerEvents> {
 export class FluidContainer extends TypedEventEmitter<IFluidContainerEvents> implements IFluidContainer {
     private readonly connectedHandler = () => this.emit("connected");
     private readonly disconnectedHandler = () => this.emit("disconnected");
-    private readonly disposedHandler = () => this.emit("disposed");
+    private readonly disposedHandler = (error?: ICriticalContainerError) => this.emit("dispose", error);
```

Comment:
What other alternatives did you consider?  Couple ideas that may or may not be good ones:

"dispose" -- the container is permanently disabled, normal exit
"error" -- the container is permanently disabled, error exit (details in payload)

"dispose" -- the container is permanently disabled, no commentary on exit reason
"error" -- the container is permanently disabled, error exit (details in payload)

"dispose" -- the container is permanently disabled, no commentary on exit reason
.error member -- inspectable to discover error details, if any

"dispose" -- the container is permanently disabled, error details (if any) in payload
.error member -- inspectable to discover error details, if any

Reply:
Great ideas here @ChumpChief. I was thinking about this at the time when we moved this change to next. In my mental model it took precedence to align APIs with IContainer events, and see these layers above IContainer as more as a subset of (existing) functionality without diverging where not absolutely necessary. If Container ends up doing some different pattern around happy vs error close, remapping may end up looking harder to maintain and reason around. If so, maybe it would be easier to reason around what IContainer should offer (and make changes there), while still keeping IFluidContainer as subset.

Im definitely open to us making improvements here, we have another ticket that will go into structured telemetry and overally more coherent experience for the azure user. I was looking at this PR as more of a short-term effort to expose what we already have, so the user can at least react to errors. And then we use the next breaking cycle to bring in these other improvements.